internal/controller/runtime/fsm/runtime_fsm_configure_skr.go

```diff
@@ -18,12 +24,31 @@ type additionalOIDCState struct {
 	hasEmptyArray bool
 }
 
-func sFnConfigureOidc(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+const (
+	msgFailedProvisioningInfoConfigMap = "Failed to apply kyma-provisioning-info config map, scheduling for retry - %s"
+	oidcErrorMessage = "Failed to create OpenIDConnect resource. Scheduling for retry"
+	kymaNamespaceCreationErrorMessage = "Failed to create kyma-system namespace. Scheduling for retry"
+)
+
+func sFnConfigureSKR(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	kymaNsCreationErr := createKymaSystemNamespace(ctx, m, s)
+	if kymaNsCreationErr != nil {
+		updateConditionFailed(&s.instance, imv1.ConditionReasonKymaSystemNSError, kymaNamespaceCreationErrorMessage)
```

Comment:
Probably we should return with updateStatusAndRequeue() or updateStatusAndRequeueAfter(m.ControlPlaneRequeueDuration)  here

```suggestion
		updateConditionFailed(&s.instance, imv1.ConditionReasonKymaSystemNSError, kymaNamespaceCreationErrorMessage)
	return updateStatusAndRequeueAfter(m.ControlPlaneRequeueDuration)
```